common/flatpak-installation.c

```diff
@@ -1221,11 +1222,41 @@ flatpak_installation_list_remotes_by_type (FlatpakInstallation     *self,
   if (!flatpak_dir_maybe_ensure_repo (dir_clone, cancellable, error))
     return NULL;
 
+#if OSTREE_CHECK_VERSION (2018, 9)
+  OstreeRepo *repo = flatpak_dir_get_repo (dir_clone);
+  if (repo != NULL)
+    default_repo_finders = ostree_repo_get_default_repo_finders (repo);
+#endif
+
+  /* If NULL or an empty array of types is passed then we use the default set
+   * provided by ostree, or fall back to using all */
   for (i = 0; i < NUM_FLATPAK_REMOTE_TYPES; ++i)
     {
-      /* If NULL or an empty array of types is passed then we include all types */
-      types_filter[i] = (num_types == 0) ? TRUE : FALSE;
+      if (num_types != 0)
+        types_filter[i] = FALSE;
+#if OSTREE_CHECK_VERSION (2018, 9)
+      else if (default_repo_finders == NULL)
+        types_filter[i] = TRUE;
+#endif
+    }
+
+#if OSTREE_CHECK_VERSION (2018, 9)
+  if (default_repo_finders != NULL && num_types == 0)
+    {
+      g_autofree char *default_repo_finders_str = g_strjoinv (" ", default_repo_finders);
+      g_debug ("Using default repo finder list: %s", default_repo_finders_str);
+
+      if (g_strv_contains (default_repo_finders, "config"))
+        types_filter[FLATPAK_REMOTE_TYPE_STATIC] = TRUE;
+      else if (g_strv_contains (default_repo_finders, "lan"))
```

Comment:
The fact that this is a non-iterated chain of `else if`s over the `default_repo_finders` means that only one o the `types_filter` elements will ever be set to `TRUE`. I don’t think that’s what you wanted?

Reply:
You're right that was stupid of me